src/darwin/Framework/CHIP/ServerEndpoint/MTRServerCluster_Internal.h

```diff
@@ -14,17 +14,15 @@
  *    limitations under the License.
  */
 
-#import <Foundation/Foundation.h>
-
-#import <Matter/MTRDeviceController.h>
 #import <Matter/MTRServerCluster.h>
 
 #include <lib/core/DataModelTypes.h>
-
 // TODO: These attribute-*.h and Span bits are a hack that should eventually go away.
 #include <app/util/attribute-metadata.h>
 #include <lib/support/Span.h>
 
+@class MTRDeviceController;
```

Comment:
Why? Including the public header seems strictly better than forward-declaring, and we know public headers can't depend on private, so dep loops are not an issue.

Reply:
We seem to forward-declare in quite a few places, and generally it's a useful way to reduce incremental build times, but if we have some general rule to always import instead of forward declaring that's fine by me.

Reply:
I would say we should import instead of forward-declaring except in cases where dependency loops would make that impossible.